modules/chatops4s-slack/src/main/scala/chatops4s/slack/SlackGateway.scala

```diff
@@ -3,20 +3,18 @@ package chatops4s.slack
 import cats.effect.{IO, Resource}
 import chatops4s.{InboundGateway, OutboundGateway}
 import chatops4s.slack.models.SlackConfig
-import org.http4s.server.Server
-import sttp.client4.cats.CatsBackend
+import sttp.client4.Backend
 import sttp.client4.httpclient.cats.HttpClientCatsBackend
 
 object SlackGateway {
 
-  def create(config: SlackConfig): Resource[IO, (OutboundGateway, InboundGateway, Server)] = {
+  def create(config: SlackConfig): Resource[IO, (OutboundGateway, InboundGateway)] = {
     for {
-      backend <- HttpClientCatsBackend.resource[IO]()
-      slackClient = new SlackClient(config, backend)
+      backend         <- HttpClientCatsBackend.resource[IO]()
```

Comment:
You're still using a concrete backend. The backend should be provided by the client (alongside config) so that they can chose whatever they want. 

(And `httpclient` dep should be removed from the module)